Plugin/Sales/Order/View/Details.php

```diff
@@ -0,0 +1,95 @@
+<?php
+declare(strict_types=1);
+
+namespace Amwal\Payments\Plugin\Sales\Order\View;
+
+use GuzzleHttp\Exception\GuzzleException;
+use Magento\Sales\Block\Adminhtml\Order\View;
+use Magento\Backend\Model\Auth\Session;
+use Magento\Framework\UrlInterface;
+use Magento\Sales\Api\Data\OrderInterface;
+use Magento\Framework\App\RequestInterface;
+use Amwal\Payments\Model\AmwalClientFactory;
+use Amwal\Payments\Model\Config;
+
+class Details
+{
+    protected $authSession;
+    protected $urlBuilder;
+    protected $order;
+    protected $request;
+    protected $amwalClientFactory;
+    private $config;
```

Comment:
It's a good practice to add type definition (added in PHP 7.4):
```suggestion
    protected Session $authSession;
    protected UrlInterface $urlBuilder;
    protected OrderInterface $order;
    protected RequestInterface $request;
    protected AmwalClientFactory $amwalClientFactory;
    private Config $config;
```